morf-h2/src/main/java/org/alfasoftware/morf/jdbc/h2/H2.java

```diff
@@ -62,6 +64,8 @@ public String formatJdbcUrl(JdbcUrlElements jdbcUrlElements) {
         .append(jdbcUrlElements.getPort() == 0 ? "" : ":" + jdbcUrlElements.getPort())
         .append("/mem:") // this means we're going to use a remote in-memory DB which isn't ideal
         .append(jdbcUrlElements.getDatabaseName());
+
+        builder.append(";INIT=CREATE SCHEMA IF NOT EXISTS " + firstNonNull(jdbcUrlElements.getSchemaName(), jdbcUrlElements.getDatabaseName()));
```

Comment:
Shouldn't we remember which schema name have we created? SchemaName or DatabaseName?

Reply:
When connecting to H2 over TCP we can either creates the schema explicitly or allow H2 to create it at init. We opted to the second approach.